lib/tree.js

```diff
@@ -12,47 +12,132 @@ class Tree {
         this.root = null;
     }
 
-    // Time Complexity: ?
-    // Space Complexity: ?
+    add_helper(current, new_node) {
+        if (current == null) { return new_node }
+
+        if (new_node.key <= current.key) { //go left
+            current.left = this.add_helper(current.left, new_node)
+        } else {
+            current.right = this.add_helper(current.right, new_node)
+        }
+        return current
+    }
+
+    // Time Complexity: O(n)
+    // Space Complexity: O(1)
     add(key, value) {
-        throw new Error("This method hasn't been implemented yet!");
+        let new_node = new TreeNode(key,value)
+        this.root = this.add_helper(this.root, new_node)
     }
 
-    // Time Complexity: ?
-    // Space Complexity: ?
+
+    find_helper(current, key) {
+        if (current == null) return null
+        if (current.key == key) return current.value
+
+        if (key < current.key && current.left != null) { //go left
+            current = current.left
+        } else if(current.right != null) {
+            current = current.right
+        } 
+        return this.find_helper(current, key)
+    }
+
+    // Time Complexity: O(n) since tree not guaranteed balanced, might have to visit all nodes
+    // Space Complexity: O(1)
     find(key) {
```

Comment:
👍 , but time & space are O(log n) if the tree is balanced and O(n) if the tree is not balanced.  

You have space complexity because the helper method are recursive.